### PR TITLE
Transitions for router changes, closes #38

### LINKED ISSUE #38: Create transitions for router changes
As user navigates, page content should be nicely transitioned (mobile + desktop).
https://material.io/design/motion/the-motion-system.html#fade-through

Let's see if we can implement something similar with https://material-ui.com/components/transitions/#transitions

### PR DESCRIPTION
**Pull request by sunny-github-acc**
Added fade transitions for router changes. Also added fade transitions for nav elements on page load.

### CODE REVIEW

File: src/pages/Home.tsx [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
 
 function Home() {
     return (
-        <Page>
-            <AppLogoHeader />
-            <RecentAnimalsContainer />
-            <RecentEventsContainer />
-        </Page>
+        <Grow in timeout={800}>
+            <div>
```
Comment (saulyz) [inline src/pages/Home.tsx:12]: why we need additional `div` here?
Documentation doesn't say a word about this need.

File: src/pages/AnimalsPage.tsx [hunk 2]
```diff
@@ -56,22 +56,26 @@ function AnimalsPage() {
...
-                    filters={filters}
-                    onFiltersClear={handleClearAllFilters}
-                    onFiltersApply={handleApplyFilters}
-                    onFilterRemove={handleRemoveFilter}
-                />
-            }
-        >
-            <AnimalsListContainer viewType={viewType} />
-        </Page>
+        <Grow in timeout={600}>
```
Comment (saulyz) [inline src/pages/AnimalsPage.tsx:59]: I would recommend changing transitions to "Fade". As in some pages content is being added after async fetches and this results in delayed paint. So when Grow transition progresses, some changes are added to the page. This breaks the visual flow.
Fade transition may help smoothen async things and it is more natural, unnoticeable. Which is good.

**Review by saulyz — CHANGES_REQUESTED** [review]
Does `AnimalDetailsPage` has transitions?
Please make some improvements and we could merge this PR.

File: src/pages/Home.tsx (hunk 1 above)
Comment (sunny-github-acc) [reply src/pages/Home.tsx]: We don't. Removed it.

File: src/pages/AnimalsPage.tsx (hunk 2 above)
Comment (sunny-github-acc) [reply src/pages/AnimalsPage.tsx]: Noted. Changed.

**Review by saulyz — APPROVED** [review]

**Review by saulyz — APPROVED** [review]

**Review by saulyz — APPROVED** [review]

**Review by AzisK — APPROVED** [review]

**Pull request merged**